dagmc/dagnav.py

```diff
@@ -289,10 +300,56 @@ def __init__(self, model: DAGModel, handle: np.uint64):
         super().__init__(model, handle)
         self._check_category_and_dimension()
 
-    def get_volumes(self):
+    @property
+    def surf_sense(self) -> list[Optional[Volume]]:
+        """Surface sense data."""
+        handles = self.model.mb.tag_get_data(
+            self.model.surf_sense_tag, self.handle, flat=True
+        )
+        return [Volume(self.model, handle) if handle != 0 else None
+                for handle in handles]
+
+    @surf_sense.setter
+    def surf_sense(self, volumes: list[Optional[Volume]]):
+        if len(volumes) != 2:
+            raise ValueError("surf_sense should be a list of two volumes.")
+        sense_data = [vol.handle if vol is not None else np.uint64(0)
+                      for vol in volumes]
+        self._tag_set_data(self.model.surf_sense_tag, sense_data)
+
+        # Establish parent-child relationships
+        for vol in volumes:
+            if vol is not None:
+                self.model.mb.add_parent_child(vol.handle, self.handle)
+
+    @property
+    def forward_volume(self) -> Optional[Volume]:
+        """Volume with forward sense with respect to the surface."""
+        try:
+            return self.surf_sense[0]
+        except RuntimeError:
```

Comment:
What's the scenario in which `surf_sense` would raise a `RuntimeError`?

Reply:
I think I ran into this before the GEOM_SENSE_2 tag had been set at all for a given surface. Looking at it now, I feel like this check could be part of the `surf_sense` getter though so let me see if I can fix that up.

Reply:
Ok, I moved the `try/except` block into the getter for `surf_sense`, which looks a lot better!